### PR TITLE
TEET-1802 - Browser tests for contracts view and edit

### PR DESCRIPTION
**Pull request by erno**

### CODE REVIEW

File: ci/browser-tests/cypress/integration/contract_mm.ci.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+context("Contracts MM", () => {
+    beforeEach(() => {
+        cy.dummyLogin("benjamin");
+        cy.selectLanguage("#EN");
+    });
+
+    it("finds contracts with search and navigates there + edits details + project link", () => {
```
Comment (erno) [inline ci/browser-tests/cypress/integration/contract_mm.ci.spec.js:7]: I changed the .click() ones to .click({force: true}) in the followon PR (https://github.com/solita/mnt-teet/pull/702)

File: app/backend/dev-src/teet/browser_test.clj [hunk 2]
```diff
@@ -1,11 +1,65 @@
...
+(defn import-test-data! []
+  (import-projects-test-data! (environment/datomic-connection))
+
+  ;; tasks import doesn't seem to yield tasks,
+  ;; but not needed either for contracts to appear.
+  ;; should investigate if we want to get tasks in browser tests too
+  (import-tasks-test-data! (environment/datomic-connection))
+
+  (import-contracts-test-data! (environment/datomic-connection))
+)
```
Comment (aleksievelizhanin) [inline app/backend/dev-src/teet/browser_test.clj:55]: outstanding bracket. Should be on the prev line.

File: app/backend/dev-src/teet/browser_test.clj [hunk 3]
```diff
@@ -1,11 +1,65 @@
...
+  [conn]
+  (let [csv-data  (.getBytes (slurp "test/teet/thk/thk-test-data.csv"))
+        projects
+        (thk-import/parse-thk-export-csv
+         {:input (java.io.ByteArrayInputStream.
+                  csv-data)
+          :column-mapping thk-mapping/thk->teet-project
+          :group-by-fn #(get % :thk.project/id)})
+        import-tasks-result (thk-import/import-thk-tasks! conn "test://test-csv" projects)]
+    ;; (println "projects was" projects)
```
Comment (aleksievelizhanin) [inline app/backend/dev-src/teet/browser_test.clj:32]: commented code, not needed -> should be removed.

File: app/backend/src/clj/teet/thk/thk_import.clj [hunk 4]
```diff
@@ -498,5 +503,7 @@
 
 (defn import-thk-tasks! [connection url projects]
   (let [db (d/db connection)]
+    ;; (println "transacting thk tasks:")
+    ;; (clojure.pprint/pprint (thk-import-tasks-tx db url projects))
```
Comment (aleksievelizhanin) [inline app/backend/src/clj/teet/thk/thk_import.clj:507]: Commented code. To be removed.

File: app/frontend/src/cljs/teet/project/project_controller.cljs [hunk 5]
```diff
@@ -588,6 +588,8 @@
                            (conj open-types :selected))]
       (assoc-in app [:route :project :open-types] new-open-types)))
 
+
+
```
Comment (aleksievelizhanin) [inline app/frontend/src/cljs/teet/project/project_controller.cljs:592]: Extra lines, are they needed?

File: app/frontend/src/cljs/teet/snackbar/snackbar_view.cljs [hunk 6]
```diff
@@ -42,6 +42,11 @@
    [SnackbarContent
     {:style {:background-color (snack-color variant)
              :flex-grow 0}
+     :data-cy (str "snackbar-" (if (keyword? variant)
+                                 (name variant)
+                                 ;; else
+                                 (pr-str variant)
+                                 ))
```
Comment (aleksievelizhanin) [inline app/frontend/src/cljs/teet/snackbar/snackbar_view.cljs:49]: Brackets outstanding, should be on the prev line.

File: ci/browser-tests/cypress/integration/contract_mm.ci.spec.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        cy.visit("#/contracts");
+        cy.get("[data-cy='search-shortcut-all-contracts']").click();
+        // cy.get("[data-cy='toggle-filters-visibility']").click(); // uncomment if default is hidden, has been in flux
+        cy.get("input[id='contract-filter-input/:contract-name']").type("TEPPO");
+
+        // cy.get("[data-cy='expand-contracts']").click({force: true});
+        // cy.get("[data-cy='expand-contracts'] span").first().click();
```
Comment (aleksievelizhanin) [inline ci/browser-tests/cypress/integration/contract_mm.ci.spec.js:14]: Commented code, if it doesn't have any purpose (explaining something, for instance) then to be removed.

**Review by aleksievelizhanin — APPROVED** [review]
Minor cosmetic cleanup suggested, other than that looks legit.

**Pull request merged**